crates/uv/src/commands/project/lock.rs

```diff
@@ -319,12 +319,12 @@ async fn do_lock(
     } else {
         let default =
             RequiresPython::greater_than_equal_version(&interpreter.python_minor_version());
-        if workspace.only_virtual() {
-            debug!("No `requires-python` in virtual-only workspace. Defaulting to `{default}`.");
-        } else {
+        if workspace.any_project() {
             warn_user!(
                 "No `requires-python` value found in the workspace. Defaulting to `{default}`."
             );
+        } else {
+            debug!("No `requires-python` in non-project workspace. Defaulting to `{default}`.");
```

Comment:
Do we even need a special message here? The above message looks fine.

Reply:
We do because there's no way to define a `requires-python` in a non-project workspace :(

Reply:
Isn't it just that you define it as a project? I still don't quite see what value the separate message adds, but don't feel strongly since it's just debug info.

Reply:
I guess we can do that now, since these are "legacy".

Reply:
(Before it was a problem because there was no way to resolve it.)

Reply:
I see, thanks!